### ISSUE TITLE
Hyperparameter importance is wrong for `maximize`d objectives

### ISSUE THREAD

**Issue by sliedes** [optuna/optuna-dashboard#1329]
### Description

Tested on both 0.20.0 and current HEAD (df2a5df73eea4c8bdc7da19055034394f8803a05); optuna 4.9.0.

When the objective function direction is `maximize`, optuna-dashboard shows wrong values for the importances. It seems it finds what is special in the _bottom_ 10% of trials instead of the top 10%.

### How to Reproduce

Consider this code. It provides two objectives, `score` and `loss`, the first being maximized and the second minimized, with `score = -loss`. Clearly the importances of hyperparameters for `score` and `loss` should be the same.

```
from pathlib import Path

import optuna


DB_PATH = Path("importance_direction_bug.sqlite3")
STORAGE = f"sqlite:///{DB_PATH}"


def objective(trial: optuna.Trial) -> tuple[float, float]:
    x = trial.suggest_float("x", 0.0, 1.0)
    y = trial.suggest_float("y", 0.0, 1.0)

    # Deliberately asymmetric:
    #
    # - good trials are characterized mostly by small x
    # - very bad trials are characterized mostly by large y
    #
    # Thus analyzing the wrong end of the objective distribution makes the
    # parameter-importance result obviously different.
    loss = x + 10.0 * y**20
    score = -loss

    return loss, score


def main() -> None:
    DB_PATH.unlink(missing_ok=True)

    study = optuna.create_study(
        study_name="importance-direction-reproducer",
        storage=STORAGE,
        directions=["minimize", "maximize"],
        sampler=optuna.samplers.RandomSampler(seed=0),
    )
    study.set_metric_names(["loss", "score"])
    study.optimize(objective, n_trials=100)


if __name__ == "__main__":
    main()
```

However, when optuna-dashboard is pointed to the database created by this, it shows this:

<img width="1378" height="625" alt="Image" src="https://github.com/user-attachments/assets/bfc84faa-ac4f-437b-b754-e0369a9898c9" />

### Python version

3.14

### Optuna version

4.9.0

### optuna-dashboard version or git revision

0.20.0; also current HEAD at df2a5df73ee (both exhibit this behavior)

### Web browser

Firefox

**Comment by kAIto47802** [comment]
Thank you for using optuna-dashboard and for reporting the issue :octocat: 

The result shown in the attached screenshot appears to be the expected result. Given:

```python
loss = x + 10.0 * y**20
score = -loss
return loss, score
```

with

```python
directions=["minimize", "maximize"]
```

the two objectives select exactly the same set of good trials. Thus, as you correctly noted, the parameter importances for score and loss should be identical.

The attached screenshot appears to show exactly this expected behavior: both objectives have approximately x = 0.98 and y = 0.02.

Therefore, I do not see an issue with the current implementation based on this result. I also reviewed the current parameter-importance calculation code and confirmed that it correctly accounts for the study direction.



The incorrect handling of `maximize` objectives was fixed in

- https://github.com/optuna/optuna-dashboard/pull/1303,

which was merged after v0.20.0. Therefore, I would expect v0.20.0 to exhibit the incorrect behavior.

Since the screenshot appears to show the fixed and expected behavior, it may have been captured when running the current HEAD. It is possible that the results obtained with v0.20.0 and the current HEAD were inadvertently mixed up.

**Comment by sliedes** [comment]
Oops, I think you're right; I tried it on both and quickly judged them the same, but I think I was mistaken! I first observed the bug on 0.20.0 and then thought I replicated it on HEAD, from which the screenshot is. But indeed, they are different and the shown result is the correct one. 🙈 Sorry for the noise!
